sdk/consumption/Azure.ResourceManager.Consumption/tests/Scenario/ConsumptionBudgetTests.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using Azure.ResourceManager.Consumption.Models;
+using Azure.ResourceManager.Resources;
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.Consumption.Tests
+{
+    internal class ConsumptionBudgetTests : ConsumptionManagementTestBase
+    {
+        private ResourceGroupResource _resourceGroup;
+        private ConsumptionBudgetCollection _consumptionBudgetCollection;
+
+        public ConsumptionBudgetTests(bool isAsync) : base(isAsync)
+        {
+        }
+
+        [SetUp]
+        public async Task TestSetUp()
+        {
+            _resourceGroup = await CreateResourceGroup();
+            _consumptionBudgetCollection = Client.GetConsumptionBudgets(scope: _resourceGroup.Id);
+        }
+
+        private async Task<ConsumptionBudgetResource> CreateBudget(string budgetName)
+        {
+            var startOn = new DateTimeOffset(Recording.Now.Year, Recording.Now.Month, 1, 0, 0, 0, new TimeSpan(0, 0, 0));
+            var data = new ConsumptionBudgetData()
+            {
+                Amount = 100,
+                TimeGrain = BudgetTimeGrainType.Monthly,
+                TimePeriod = new BudgetTimePeriod(startOn)
+                {
+                    EndOn = startOn.AddYears(1)
+                },
+                Category = BudgetCategory.Cost,
+            };
+            var budget = await _consumptionBudgetCollection.CreateOrUpdateAsync(WaitUntil.Completed, budgetName, data);
+            return budget.Value;
+        }
+
+        [RecordedTest]
+        public async Task CreateOrUpdate()
+        {
+            string budgetName = Recording.GenerateAssetName("budget");
+            var budget = await CreateBudget(budgetName);
```

Comment:
Consider moving `CreateBudget` to `Setup` since it's needed for all tests except the one in `Delete`.